pages/en/zkapps/how-to-write-a-zkapp.mdx

```diff
@@ -391,14 +391,22 @@ Off-chain state is not yet available.
 
 ### Testing your smart contract
 
-Testing is critical when writing smart contracts.
+Writing automated tests for your smart contract is of crucial importance. The <a href="https://jestjs.io/">Jest testing framework</a> is included in all projects created by the Mina zkApp CLI via `zk project <name>`. We recommend Jest as your project's test runner, but any testing framework can be used.
 
-The <a href="https://jestjs.io/">Jest testing framework</a> is included in all projects created by the Mina zkApp CLI via `zk project <name>`. We recommend Jest, but any testing framework can be used.
+##### Running tests
 
 To run all test files within your project, run `npm run test` or `npm run testw` (for watch mode) from your project’s root directory.
 
 To generate a test coverage report for your project, run `npm run coverage`. Coverage reports show the % of your code that is tested. The result will be output in your terminal. This can be helpful to ensure your code is well tested.
 
+##### Writing tests
+
+To write a test for your smart contract, create a new TypeScript file with the naming convention `your_contract_name.test.ts`. This is the file where you should write all your smart contract test code. To write good unit tests, it's vital that you concretely understand the functionality your smart contract provides.
+
+Because we are using Jest as our default test runner, it's helpful to break down all functionality of your smart contract into `describe` blocks. Mapping out your unit tests like this is also a good way of providing documentation to other developers reading your smart contract. An excellent place to start testing your smart contract is in the areas your smart contract modifies its state. Make sure to verify that your state updates in the way you expect.
+
```

Comment:
I think it would be helpful for users if we very briefly mention the use of the Mina.LocalBlockchain() method used for testing. Something similar to the comments in the example project index.ts file for running  projects locally would be ok followed by a code snippet.

The `Mina.LocalBlockchain()` method specifies a ledger of accounts and contains logic for updating the ledger that can be used to test your smart contract locally. 

```
function createLocalBlockchain() {
  const Local = Mina.LocalBlockchain();
  Mina.setActiveInstance(Local);
  return Local.testAccounts[0].privateKey;
}
```



 * Please note that this deployment is local and does not deploy to a live network.
 * If you wish to deploy to a live network, please use the zkapp-cli to deploy.

Reply:
Could we intro by saying that when they create a file using `zk file foo`, they will `foo.ts` and `foo.test.ts`. Goal is to reinforce intended usage of our CLI and providing the file naming example within the text helps to establish our recommended file naming convention too.

Might be good to add a basic, _empty_ example contents of a test file showing an import and a couple nested describe statements with the file name and method name, and one `it()` test where the contents are literally just a JS comment saying something like `// your test here`.  

Then I really like your guidance for what they should look to test w/r/t state updates. +1 for Yoni's suggestion to mention link it to the Add smart contract, as a simple example showing state + 2.